core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java

```diff
@@ -3301,6 +3301,85 @@ private void checkHavingAliasSameAsColumn(boolean upperAlias) {
     sql(query).ok(expected);
   }
 
+  @Test void testSetOpLimitShouldRetainParentheses() {
```

Comment:
There are only tests for UNION currently. Add a test for INTERSECT or EXCEPT, or perhaps a combination of set-ops.

Reply:
There has been a case of UNION and INTERSECT before, I've added a combination of set-ops.